libraries/docker_image.rb

```diff
@@ -16,12 +16,37 @@ class DockerImage < DockerBase
     property :rm, [true, false], default: true
     property :source, String
     property :tag, String, default: 'latest'
+    property :buildargs, [String, Array, Hash], coerce: proc { |v| coerce_buildargs(v) }
 
     alias_method :image, :repo
     alias_method :image_name, :repo
     alias_method :no_cache, :nocache
     alias_method :no_prune, :noprune
 
+    ###################
+    # Property helpers
+    ###################
+
+    def coerce_buildargs(v)
+      case v
+      when Hash
+        str = ''
+        v.each do |key, value|
+          str += '"' + key.to_s + '": "' + value.to_s + '",'
```

Comment:
As below but something like this instead:
```suggestion
          v.map { |k, v| "\"#{k}\": \"#{v}\""}.join(',')
```